js/gateio.js

```diff
@@ -2420,53 +2421,137 @@ module.exports = class gateio extends Exchange {
+        //        "reason": "",
+        //        "create_time": 1643953482,
+        //        "finish_time": 1643953482,
+        //        "is_stop_order": false,
+        //        "stop_trigger": {
+        //            "rule": 0,
+        //            "trigger_price": "",
+        //            "order_price": ""
+        //        },
+        //        "me_order_id": 0,
+        //        "order_type":""
+        //    }
+        //
+        const put = this.safeValue2 (order, 'put', 'initial');
+        const trigger = this.safeValue (order, 'trigger');
+        // if (put) {
+        let contract = this.safeString (put, 'contract');
+        let type = this.safeString (put, 'type');
+        let timeInForce = this.safeStringUpper2 (put, 'time_in_force', 'tif');
+        let amount = this.safeString2 (put, 'amount', 'size');
+        let side = this.safeString (put, 'side');
+        let price = this.safeString (put, 'price');
+        // } else {
+        contract = this.safeString (order, 'contract', contract);
+        type = this.safeString (order, 'type', type);
+        timeInForce = this.safeStringUpper2 (order, 'time_in_force', 'tif', timeInForce);
+        amount = this.safeString2 (order, 'amount', 'size', amount);
+        side = this.safeString (order, 'side', side);
+        price = this.safeString (order, 'price', price);
+        let remaining = this.safeString (order, 'left');
+        let filled = Precise.stringSub (amount, remaining);
+        let cost = this.safeNumber (order, 'filled_total');
+        if (put) {
+            remaining = amount;
+            filled = '0';
+            cost = this.parseNumber ('0');
+        }
+        // }
         if (contract) {
-            // fill price is the price per contract for swaps, but the cost for spot
-            average = this.safeString (order, 'fill_price');
-            if (amount) {
-                side = Precise.stringGt (amountRaw, '0') ? 'buy' : 'sell';
-            } else {
-                side = undefined;
-            }
-            rawStatus = this.safeString (order, 'finish_as', 'open');
-        } else {
-            // open, closed, cancelled - almost already ccxt unified!
-            rawStatus = this.safeString (order, 'status');
-            side = this.safeString (order, 'side');
-        }
-        const status = this.parseOrderStatus (rawStatus);
-        let timeInForce = this.safeStringUpper2 (order, 'time_in_force', 'tif');
-        if (timeInForce === 'POC') {
-            timeInForce = 'PO';
-        }
-        let type = this.safeString (order, 'type');
-        if (type === undefined) {
-            // response for swaps doesn't include the type information
-            if (timeInForce === 'PO' || timeInForce === 'GTC' || timeInForce === 'IOC' || timeInForce === 'FOK') {
-                type = 'limit';
-            } else {
-                type = 'market';
-            }
+            type = (timeInForce === 'IOC') ? 'market' : 'limit';
```

Comment:
Are you sure that IOC == market ? Cause as far as I understand, IOC is for limit orders under a specific price, while market orders don't have a price threshold.

Reply:
![](https://i.imgur.com/3D8o6qf.png)